### DISCUSSION TITLE
Is it possible to modify incoming c.req?

### DISCUSSION THREAD

**Discussion by nedimaf** [honojs/hono#3484] [Q&A]
I have a middleware which should attach new header to the specific incoming requests and pass them further. It seems that `c.req.header` can only get the existing header, but not modify them. Also, `c.headers` can add new header, but not to the specific request. Is there any alternative for headers modification?

**Comment by yusukebe** [comment]
Hi @nedimaf 

You should create a new `Request` object to modify headers. If you want to use it in `c.req`, you can write like the following hack-way:

```ts
app.get('/', (c) => {
  const req = new Request(c.req.raw)
  req.headers.append('foo', 'bar')
  c.req.raw = req
  return c.text(c.req.header('foo'))
})
```

**Reply by nedimaf** [reply]
@yusukebe that works! Thank you very much :)

**Reply by sant123** [reply]
@yusukebe thanks a lot!!
